community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/steps/planShortestPaths.scala

```diff
@@ -36,8 +36,12 @@ case object planShortestPaths {
            (implicit context: LogicalPlanningContext): LogicalPlan = {
 
     val variables = Set(shortestPaths.name, Some(shortestPaths.rel.name)).flatten
+    def predicateAppliesToShortestPath(p: Predicate) =
```

Comment:
Can we have a comment here explaining the logic of the below rows? I don't think I fully understand it now and future me will surely be clueless.